package/network/utils/comgt/files/ncm.sh

```diff
@@ -117,21 +127,23 @@ proto_ncm_setup() {
 	}
 
 	echo "Starting network $interface"
-	json_get_vars connect
-	eval COMMAND="$connect" gcom -d "$device" -s /etc/gcom/runcommand.gcom || {
-		echo "Failed to connect"
-		proto_notify_error "$interface" CONNECT_FAILED
-		return 1
-	}
-
 	echo "Setting up $ifname"
-	
+
 	proto_init_update "$ifname" 1
 	proto_add_data
 	json_add_string "manufacturer" "$manufacturer"
 	proto_close_data
 	proto_send_update "$interface"
 
+	json_load "$(cat /etc/gcom/ncm.json)"
+	json_select "$manufacturer"
+	json_get_vars connect
+	eval COMMAND="$connect" gcom -d "$device" -s /etc/gcom/runcommand.gcom || {
+		echo "Failed to connect"
+		proto_notify_error "$interface" CONNECT_FAILED
+		return 1
+	}
+
```

Comment:
Thank you. I think that the modem firmware indeed crashes in the first case (@bmork any idea what could cause this?).

I don't think that we have any other way of working around this, so here's what I'd propose: Since AT+CGACT is actually the command that connects the modem, make that the "connect" command in ncm.json. Then move AT+CGDCONT to "initialize" list and we don't need the "configure" list anymore. Finally, add a new "finally" (or any better name you can think of) command, which is executed at the end. Store it in a variable before `proto_init_update "$ifname" 1`, so you don't need to load the json file again afterwards.